modules/social_features/social_event/src/EdaHandler.php

```diff
@@ -109,4 +171,45 @@ public function fromEntity(NodeInterface $node): CloudEvent {
     );
   }
 
+  /**
+   * Determines the actor (application and user) for the CloudEvent.
+   *
+   * @return array
+   *   An array with two elements: the application and the user.
+   */
+  private function determineActors(): array {
+    $application = NULL;
+    $user = NULL;
+
+    switch ($this->routeName) {
+      case 'entity.node.edit_form':
+      case 'system.admin_content':
+        $user = $this->currentUser;
+        break;
+
+      case 'entity.ultimate_cron_job.run':
+        $application = 'cron';
+        break;
+    }
+
+    return [
+      $application,
+      $user,
+    ];
+  }
+
+  /**
+   * Dispatches the event.
+   *
+   * @param string $topic_name
+   *   The topic name.
+   * @param \CloudEvents\V1\CloudEvent $event
+   *   The CloudEvent event.
+   *
+   * @return void
```

Comment:
@tregismoreira I see [PHPCS](https://github.com/goalgorilla/open_social/actions/runs/11167591676/job/31044319498?pr=4104) check failed on `Description for the @return value is missing`

Reply:
@ribel fixed. Thanks ;)